mongomock/aggregate.py

```diff
@@ -543,6 +544,41 @@ def _handle_type_convertion_operator(self, operator, values):
                 raise TypeError("'%s' type is not supported" % type(parsed))
             return decimal_value
 
+        # Document: https://docs.mongodb.com/manual/reference/operator/aggregation/arrayToObject/
+        if operator == '$arrayToObject':
+            try:
+                parsed = self.parse(values)
+            except KeyError:
+                return None
+
+            if parsed is None:
+                return None
+
+            if not isinstance(parsed, list):
+                raise TypeError('$arrayToObject only supports arrays')
+
+            def check_doc(elements):
+                return all([isinstance(x, dict) and 'k' in x and 'v' in x and len(x) == 2
```

Comment:
I don't think you need the `[]` here; Also you might be able to check with set equality:
```py
return all(isinstance(x, dict) and x.keys() == {'k', 'v'} for x in elements)
```